layers/+distributions/spacemacs-base/funcs.el

```diff
@@ -184,6 +184,49 @@ Dedicated (locked) windows are left untouched."
   (interactive "p")
   (spacemacs/rotate-windows (* -1 count)))
 
+(defun spacemacs/move-buffer-to-window (windownum follow-focus-p)
+  "Moves a buffer to a window, using the spacemacs numbering. follow-focus-p
+   controls whether focus moves to new window (with buffer), or stays on
+   current"
+  (interactive)
+  (let ((b (current-buffer))
+        (w (selected-window)))
+    (save-selected-window
```

Comment:
There isn't really a separate get-window-by-number function defined.

```
(defun select-window-by-number (i &optional arg)
  "Select window given number I by `window-numbering-mode'.
If prefix ARG is given, delete the window instead of selecting it."
  (interactive "P")
  (let ((windows (car (gethash (selected-frame) window-numbering-table)))
        window)
    (if (and (>= i 0) (< i 10)
             (setq window (aref windows i)))
        (if arg
            (delete-window window)
          (select-window window))
      (error "No window numbered %s" i))))
```

I could either repeat the first line where I call gethash, and do bounds checking. Or (my preference), if the problem is only with the `switch-to-buffer call`, then I can use `select-window-by-number` followed by `selected-window` to get the window, and then call `set-window-buffer`. It's a tiny bit roundabout, but it achieves the best reuse of existing code and is robust. I'd strongly prefer the latter; let me know if you are ok with it.